public/app/plugins/datasource/mixed/MixedDataSource.ts

```diff
@@ -85,7 +85,7 @@ export class MixedDatasource extends DataSourceApi<DataQuery> {
       )
     );
 
-    return forkJoin(runningQueries).pipe(map(this.finalizeResponses), mergeAll());
+    return combineLatest(runningQueries).pipe(map(this.finalizeResponses), mergeAll());
```

Comment:
hmm, are we sure this is the way to go? Feels like the `forkJoin` should have been the correct operator for this scenario. Are we sure that the underlying observables are completed correctly?

Reply:
I agree with Marcus with `combineLatest` you need to make sure all observables have completed at least once (if I remember correctly). You need to be carful to catch any errors with `catchError` and return `EMPTY` for instance.

Reply:
Ahh, you are right. This is not the way to go. I'll take this for another spin.

Reply:
forkJoin was not good because it only emitted the last value for example:
I have 2 loki queries and one prometheus
loki --(1)--(2)--|
prometheus --(3)--|

in mixed it looked like this --(2)--(3)--| because of forkJoin but I think what would make sense is to return all the values like this --(1)--(2)--(3)--|

I've changed the code a little bit so if one of the queries returns error then the first error should be shown. This was possible by pushing it to the array but now instead of the last successful response I show the first one with error.

Reply:
Using `concatMap` means that every query has to run to completion before the next query runs. Is this really what we want?

Reply:
I changed to concatMap so we return in the same order as we have the queries. Is this not necessary?

Reply:
I get that, but instead of running all the queries in parallel (forkJoin), with `concatMap` we're running each query in serial. I was wondering if that is the expected behavior? @mckn any comments?

Reply:
So the problem is that we have datasources that has multiple queries and we only get the latest response in that scenario?